STIPS/STIP-009.md

```diff
@@ -0,0 +1,1178 @@
+|mapping(address => IDelegatedManager)|setManagers|Mapping from Set Token to DelegatedManager|
+
+#### Functions
+
+| Name  | Caller  | Description     |
+|------	|------	|-------------	|
+|initialize|owner|Initialize the TradeExtension on the DelegatedManager and initialize the TradeModule on the SetToken if necessary|
+|trade|operator|Trade between whitelisted assets on a DEX|
+
+----
+
+### Functions
+
+> initializeExtension
+
+ONLY OWNER: Initialize the TradeExtension on the DelegatedManager
+
+```solidity
+function initializeExtension(address _delegatedManager)
+```
++ require that extension state in delegatedManager's *extensionAllowList* is *PENDING*
++ set *setManagers[_setTokenAddress]* to manager
++ call `manager.initializeExtension()`
++ emit *ExtensionInitialized* event
+
+----
+
+> initializeModuleAndExtension
+
+ONLY OWNER: Initialize the TradeModule and the TradeExtension
+
+```
+function initializeModuleAndExtension(address _delegatedManager, IManagerIssuanceHook _preIssueHook)
+```
+
++ call *initializeExtension(_delegatedManager)*
++ Formulate call to initialize module from manager
+    ```solidity
+    bytes memory callData = abi.encodeWithSelector(
+        ITradeModule.initialize.selector,
+        delgatedManager.setToken()
+    );
+    ```
+
++ call *invokeManager(streamingFeeModule, callData)*
+
+### BasicIssuanceExtension
+
+#### Inheritance
+
+- BaseGlobalExtension
+
+##### FeeState
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|address|feeRecipient|Address to accrue fees to|
+|uint256|maxIssueFee|Max issuance fee manager commits to using (1% = 1e16, 100% = 1e18)|
+|uint256|issueFee|Percent of Set accruing to manager on issuance (1% = 1e16, 100% = 1e18)|
+|uint256|maxRedeemFee|Max redemption fee manager commits to using (1% = 1e16, 100% = 1e18)|
+|uint256|redeemFee|Percent of Set accruing to manager on redemption (1% = 1e16, 100% = 1e18)|
+
+#### Global Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|IIssuanceModule|issuanceModule|Issuance Module for Set Token|
+
+#### Public Variables
+
+| Type 	| Name 	| Description 	|
+|------	|------	|-------------	|
+|mapping(address => IDelegatedManager)|setManagers|Mapping from Set Token to DelegatedManager|
+
+#### Functions
+
+| Name  | Caller  | Description     |
+|------	|------	|-------------	|
+|initialize|owner|Initialize the BasicIssuanceExtension on the DelegatedManager and initialize the BasicIssuanceModule on the SetToken if necessary|
+|updateIssueFee|owner|Update issue fee on IssuanceModule|
+|updateRedeemFee|owner|Update redeem fee on IssuanceModule|
+
+----
+
+### Functions
+
+> initializeExtension
+
+ONLY OWNER: Initialize the BasicIssuanceExtension on the DelegatedManager
+
+```solidity
+function initializeExtension(address _delegatedManager)
+```
+
++ read _setTokenAddress from _delegatedManager
++ require that caller be the *deployer* specified in the factory's *initialize[_setTokenAddress]* mapping
```

Comment:
This should be the owner. Once the deployer address is used for the initial deployment it should have no other input